Remotion/Web/Development.WebTesting/HostingStrategies/IisDockerContainerWrapper.cs

```diff
@@ -28,109 +23,27 @@ namespace Remotion.Web.Development.WebTesting.HostingStrategies
   /// <summary>
   /// Represents an IIS Docker container and manages its lifecycle.
   /// </summary>
-  public class IisDockerContainerWrapper : IDisposable
+  public class IisDockerContainerWrapper : DockerContainerWrapperBase, IDisposable
   {
-    private static readonly ILog s_log = LogManager.GetLogger(typeof(IisDockerContainerWrapper));
-
-    private readonly IDockerClient _docker;
-    private readonly IisDockerContainerConfigurationParameters _configurationParameters;
-    private string? _containerName;
-
     public IisDockerContainerWrapper (
-        [NotNull] IDockerClient docker,
-        [NotNull] IisDockerContainerConfigurationParameters configurationParameters)
-    {
-      ArgumentUtility.CheckNotNull("docker", docker);
-      ArgumentUtility.CheckNotNull("configurationParameters", configurationParameters);
-
-      _docker = docker;
-      _configurationParameters = configurationParameters;
-    }
-
-    /// <summary>
-    /// Pulls and starts the configured docker image as a container with the settings specified in the App.config.
-    /// </summary>
-    public void Run ()
-    {
-      try
-      {
-        _docker.Pull(_configurationParameters.DockerImageName);
-      }
-      catch (DockerOperationException ex)
-      {
-        s_log.Error($"Pulling the docker image '{_configurationParameters.DockerImageName}' failed. Trying to proceed with a locally cached image.", ex);
-      }
-
-      var mounts = GetMountsWithWebApplicationPath(_configurationParameters.Mounts);
-      var portMappings = new Dictionary<int, int> { { _configurationParameters.WebApplicationPort, _configurationParameters.WebApplicationPort } };
-
-      _containerName = _docker.Run(
-          portMappings,
-          mounts,
-          _configurationParameters.DockerImageName,
-          _configurationParameters.DockerIsolationMode,
-          _configurationParameters.Hostname,
-          true,
-          "powershell",
-          $@"-Command ""
-C:\Windows\System32\inetsrv\appcmd.exe set apppool /apppool.name:DefaultAppPool /enable32bitapponwin64:{_configurationParameters.Is32BitProcess};
-C:\Windows\System32\inetsrv\appcmd.exe add site /name:WebTestingSite /physicalPath:""{_configurationParameters.AbsoluteWebApplicationPath}"" /bindings:""http://*:{_configurationParameters.WebApplicationPort}"";
-C:\ServiceMonitor.exe w3svc;
-""");
-    }
-
-    /// <inheritdoc />
-    public void Dispose ()
+        IDockerClient docker,
+        DockerContainerConfigurationParameters configurationParameters)
```

Comment:
I think that is useless code duplication, we only have a single usage in the docker hosting strategy, the main class where all the docker logic happens. It should be obvious enough.

Reply:
okay, I can live with it. The code is isolated and likely never used outside of our infrastructure. 

Question though: would you make the same choice if the type didn't have that many ctor parameters and it would thus be less code to create a matching inheritance hierarchy?

Rule of thumb: it typically makes code relations easier to understand and follow if there's symetric inheritance hierarchies. You see from the naming which types work together.

Reply:
I think what matters more here is that this probably will never be used outside of our infrastructure, and would just mean another file to maintain.